### PR TITLE
Remove multiple if-else statement in the get_activation function.

### PR DESCRIPTION
**Pull request by hi-sushanta**
Hello to all maintainers 🤗

I have made some minor changes to the activation file.

Replace multiple if-else statements with a dictionary of activation functions, and call one if statement to retrieve the appropriate function.

Now the function looks like this:
Screenshot from 2023-10-19 21-28-37

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by hi-sushanta** [comment]
If anyone notices my updates, I'd love to hear their thoughts.

**Review by patrickvonplaten — APPROVED** [review]
Ok for me. Wdyt @sayakpaul ?

File: src/diffusers/models/activations.py [hunk 1]
```diff
@@ -10,13 +10,17 @@ def get_activation(act_fn: str) -> nn.Module:
...
-    elif act_fn == "relu":
-        return nn.ReLU()
+
+    activation_functions = {
+        "swish": nn.SiLU(),
+        "silu": nn.SiLU(),
+        "mish": nn.Mish(),
+        "gelu": nn.GELU(),
+        "relu": nn.ReLU(),
+    }
```
Comment (sayakpaul) [inline src/diffusers/models/activations.py:20]: Could we maybe define it as a constant outside of the function?

Comment (hi-sushanta) [reply src/diffusers/models/activations.py:20]: Yes, that is a great idea.

Comment (hi-sushanta) [reply src/diffusers/models/activations.py]: OK, Now update my code.

Comment (sayakpaul) [reply src/diffusers/models/activations.py]: We need to resolve the merge conflicts first.

**Review by sayakpaul — APPROVED** [review]
Thank you! Looks clean!

Will merge after the CI passes.

**Comment by hi-sushanta** [comment]
Ok, now CI is passed.

**Pull request merged**